frontend/src/search/hooks/useSearch.ts

```diff
@@ -0,0 +1,107 @@
+import { useState, useEffect } from 'react';
+import { useSearchFoldersQuery, useSearchJournalsQuery } from '@/graphql/generated';
+
+const useSearch = () => {
+  const [searchQuery, setSearchQuery] = useState('');
+  const [isSearching, setIsSearching] = useState(false);
+  const [inputValue, setInputValue] = useState('');
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [activeFilter, setActiveFilter] = useState<string | null>(null);
```

Comment:
Whenever you have a defined selection of values, it's better to type it more strictly, instead of any string

Use either enum, object of union type

Enum:
```
enum FilterType = {
  Newest,
  Oldest,
  Folder
}


```

Object:
```
const FILTER_TYPES = {
  FOLDER: 'FOLDER'
  NEWEST: 'NEWEST
}
```

Union
```
type FilterType = 'folder' | 'newest' | 'oldest'
```


I personally prefer using enums, so I would do something like

`const [activeFilter, setActiveFilter] = useState<FilterType | null>(null);`

And then use it like `setActiveFilter(FilterType.Folder)`